solr/core/src/test/org/apache/solr/cloud/TestShardsWithSingleReplica.java

```diff
@@ -0,0 +1,134 @@
+
+  @BeforeClass
+  public static void setupCluster() throws Exception {
+    // Horrible copy-patsa.
+    System.setProperty("solr.test.sys.prop1", "propone");
+    System.setProperty("solr.test.sys.prop2", "proptwo");
+
+    final String configName = "solrCloudCollectionConfig";
+    File configDir = new File(TEST_HOME() + File.separator + "collection1" + File.separator + "conf");
+    configureCluster(3)
+        .addConfig(configName, configDir.toPath())
+        .configure();
+  }
+
+  public void testSkipLeaderOperations() throws Exception {
+    String overseerLeader = getOverseerLeader();
+    List<JettySolrRunner> notOverseerNodes = new ArrayList<>();
+    List<String> createNodeNames = new ArrayList<>();
+    for (JettySolrRunner solrRunner : cluster.getJettySolrRunners()) {
+      String nodeName = solrRunner.getCoreContainer().getZkController().getNodeName();
+      if (!nodeName.equals(overseerLeader)) {
+        notOverseerNodes.add(solrRunner);
+        createNodeNames.add(nodeName);
+      }
+    }
+
+    final String collection = "collection1";
+    new CollectionAdminRequest.Create()
+        .setCollectionName(collection)
+        .setConfigName("solrCloudCollectionConfig")
+        .setNumShards(2)
+        .setReplicationFactor(1)
+        .setCreateNodeSet(Joiner.on(',').join(createNodeNames))
+        .process(cluster.getSolrClient());
+
+    for (JettySolrRunner solrRunner : notOverseerNodes) {
+      solrRunner.stop();
+    }
+
+    waitForState("Expected single liveNode", new Predicate<ClusterState>() {
+      @Override
+      public boolean apply(ClusterState input) {
+        return input.getLiveNodes().size() == 1;
+      }
+    });
+
+    CollectionAdminResponse resp = new CollectionAdminRequest.OverseerStatus().process(cluster.getSolrClient());
+    for (JettySolrRunner solrRunner : notOverseerNodes) {
+      cluster.startJettySolrRunner(solrRunner);
+    }
+
+    waitForState("Expected 2x1 for collection: " + collection, new Predicate<ClusterState>() {
+      @Override
+      public boolean apply(ClusterState input) {
+        Set<String> liveNodes = input.getLiveNodes();
+        DocCollection collectionState = input.getCollection(collection);
+        if (collectionState == null)
+          return false;
+        if (collectionState.getSlices().size() != 2)
+          return false;
+        for (Slice slice : collectionState.getSlices()) {
+          int activeReplicas = 0;
+          for (Replica replica : slice.getReplicas()) {
+            if (liveNodes.contains(replica.getNodeName()) && replica.getState() == Replica.State.ACTIVE)
+              activeReplicas++;
+          }
+          if (activeReplicas != 1)
+            return false;
+        }
+        return true;
+      }
+    });
+
+    CollectionAdminResponse resp2 = new CollectionAdminRequest.OverseerStatus().process(cluster.getSolrClient());
+    assertEquals(getNumLeaderOpeations(resp), getNumLeaderOpeations(resp2));
```

Comment:
Assert they are both 2, one for each shard, instead?

Reply:
This section is actually identical to what landed on master.